testing/python/issue/test_tilelang_issue_2365.py

```
@@ -0,0 +1,217 @@
+
+@tilelang.testing.requires_cuda
+@tilelang.testing.requires_cuda_compute_version_ge(*_MIN_CC)
+def test_pipelined_residual_gemm_compiles_and_correct():
+    """The original codegen crash: each trigger case must compile, run, produce
+    no NaN, and match torch."""
+    failures = []
+    for name, M, N, K, bM, bN, bK, dt in _TRIGGER_CASES:
+        rtol, atol = _TOL[dt]
+        try:
+            c, ref = _compile_run(M, N, K, bM, bN, bK, dt)
+        except Exception as e:  # compile crash == the bug we are guarding
```

Ruff (0.15.15)

[warning] Do not catch blind exception: `Exception`

(BLE001)